python/langsmith/client.py

```diff
@@ -1290,6 +1325,19 @@ def create_run(
         ):
             if self._pyo3_client is not None:
                 self._pyo3_client.create_run(run_create)
+            if self.compressed_runs_buffer is not None:
+                serialized_op = serialize_run_dict("post", run_create)
+                multipart_form = (
+                    serialized_run_operation_to_multipart_parts_and_context(
+                        serialized_op
+                    )
+                )
+                with self._buffer_lock:
```

Comment:
hm.. I'm not sure what way there is around it given that it seems like it's not thread safe, but this is acquiring a lock in the main thread which feels detrimental to application runtime

Reply:
I can't see a way around this. If you view the benchmarks in the notion doc the compression minimally affects user perceived time, whereas flush time becomes significantly faster